compiler/rustc_hir/src/hir.rs

```diff
@@ -3531,7 +3526,8 @@ pub enum Node<'hir> {
     WhereBoundPredicate(&'hir WhereBoundPredicate<'hir>),
     // FIXME: Merge into `Node::Infer`.
     ArrayLenInfer(&'hir InferArg),
-    AssocOpaqueTy(&'hir AssocOpaqueTy),
+    // Created by query feeding
+    Synthetic,
```

Comment:
In https://github.com/rust-lang/rust/pull/120206 and https://github.com/rust-lang/rust/pull/120943 I specifically wanted to avoid having a non-specific "dummy" node kind.
If we have some `HirId` or `DefId` then it's better to know whether it refers to an associated opaque type, or implicit static, or some specific non-synthesized HIR node, at least for debugging.

Reply:
I can give it a field where we can encode the kind of synthetic node that we have here. Then it can be handled the same everywhere and get custom handling where desirable

We usually can get this information by asking for the `DefId` and then the `DefKind`, but that's not always immediately there I guess